datamodel_code_generator/parser/jsonschema.py

```diff
@@ -548,41 +548,59 @@ def parse_enum(
 
     def parse_ref(self, obj: JsonSchemaObject, path: List[str]) -> None:
         if obj.ref:
-            ref: str = obj.ref
-            # https://swagger.io/docs/specification/using-ref/
-            if obj.ref.startswith('#'):
-                # Local Reference – $ref: '#/definitions/myElement'
-                pass
-            elif '://' in ref:
-                # URL Reference – $ref: 'http://path/to/your/resource' Uses the whole document located on the different server.
-                raise NotImplementedError(f'URL Reference is not supported. $ref:{ref}')
+            reference = self.model_resolver.get_by_path(obj.ref)
+            if not reference or not reference.loaded:
 
-            else:
-                # Remote Reference – $ref: 'document.json' Uses the whole document located on the same server and in
-                # the same location. TODO treat edge case
-                relative_path, object_path = ref.split('#/')
-                full_path = self.base_path / relative_path
-                with full_path.open() as f:
-                    if full_path.suffix.lower() == '.json':
-                        import json
-
-                        ref_body: Dict[str, Any] = json.load(f)
-                    else:
-                        # expect yaml
-                        import yaml
+                # https://swagger.io/docs/specification/using-ref/
+                if obj.ref.startswith('#'):
+                    # Local Reference – $ref: '#/definitions/myElement'
+                    pass
+                else:
+                    relative_path, object_path = obj.ref.split('#/')
+                    if obj.ref.startswith('https://') or obj.ref.startswith('http://'):
+                        # URL Reference – $ref: 'http://path/to/your/resource' Uses the whole document located on the different server.
+                        try:
+                            import httpx
+                        except ImportError:  # pragma: no cover
+                            raise Exception(
+                                f'Please run $pip install datamodel-code-generator[http] to resolve URL Reference ref={ref}'
+                            )
+                        raw_body: str = httpx.get(relative_path).text
+                        if relative_path.lower().endswith('.json'):
+                            import json
 
-                        ref_body = yaml.safe_load(f)
+                            ref_body: Dict[str, Any] = json.loads(raw_body)
+                        else:
+                            # expect yaml
+                            import yaml
+
+                            ref_body = yaml.safe_load(raw_body)
```

Comment:
Do we need typings here too?

```suggestion
                            ref_body: Dict[str, Any] = yaml.safe_load(raw_body)
```

Reply:
OK, I will add it.

Reply:
Mypy say I should define type hint for a variable only one time.